src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -1956,7 +2026,9 @@ abstract class ManagedStrategy(
 
     private fun objectFqTypeName(obj: Any?): String {
         val enumPrefix = if (obj?.javaClass?.isEnum == true) "Enum:" else ""
-        return "$enumPrefix${obj?.javaClass?.name ?: "null"}"
+        val typeName = obj?.javaClass?.name ?: "null"
+        return if (enumPrefix.isEmpty()) typeName
```

Comment:
Looks like `if` is redundant here, just `$enumPrefix$typeName` would do the same.

Reply:
As I profiled earlier the version `"$enumPrefix$typeName"` copies the string. But in most of the invocations of this method, I just need to return `obj?.javaClass?.name` which is already allocated

Reply:
Ok, let's then wrap the entier `Enum` path into separate `if` branch:
```
private fun objectFqTypeName(obj: Any?): String {
        val typeName = obj?.javaClass?.name ?: "null"
        // Note: `if` here is important for performance reasons;
        // In common case we want to return just `typeName` without using string templates to avoid redundant string allocation.
        if (obj?.javaClass?.isEnum == true) {
            return "Enum:$typeName"
        }
        return typeName
}
```